core/src/main/scala/kafka/log/Log.scala

```diff
@@ -893,6 +926,7 @@ class Log(@volatile var dir: File,
         activeSegment.truncateTo(targetOffset)
         updateLogEndOffset(targetOffset)
         this.recoveryPoint = math.min(targetOffset, this.recoveryPoint)
+        this.logStartOffset = math.min(targetOffset, this.logStartOffset)
```

Comment:
Should it be an error to truncate beyond the logStartOffset? LogStartOffset should not decrease, right?